src/test/java/jenkins/plugins/github/api/mock/MockGitHub.java

```diff
@@ -0,0 +1,233 @@
+package jenkins.plugins.github.api.mock;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import hudson.Util;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import javax.servlet.ServletException;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.jvnet.hudson.test.ThreadPoolImpl;
+import org.kohsuke.stapler.HttpResponse;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+public class MockGitHub implements Closeable {
+    private AtomicLong nextId = new AtomicLong();
+    private Map<String, MockUser> users = new HashMap<>();
+    private Map<String, MockOrganization> organizations = new HashMap<>();
+
+    private Server server;
+
+    private int localPort = -1;
+    private JsonFactory factory = new JsonFactory();
+
+    public String open() throws IOException {
+        server = new Server(new ThreadPoolImpl(
+                new ThreadPoolExecutor(10, 10, 10L, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>(),
+                        new ThreadFactory() {
+                            public Thread newThread(Runnable r) {
+                                Thread t = new Thread(r);
+                                t.setName("Jetty Thread Pool");
+                                return t;
+                            }
+                        })));
+        ServletContextHandler context = new ServletContextHandler();
+        server.setHandler(context);
+        ;
+        context.addServlet(Stapler.class, "/*");
```

Comment:
I'm using Stapler here rather than JAX-RS because I do not want the Jackson dependency version to be coming from the JAX-RS dependency tree... so to make the tests effective I need a framework that can handle the JSON without pulling in its own version of Jackson